src/core_scan/Core_scan.ml

```diff
@@ -165,7 +165,7 @@ let replace_named_pipe_by_regular_file path =
         let remove () = if Sys.file_exists tmp_path then Sys.remove tmp_path in
         (* Try to remove temporary file when program exits. *)
         at_exit remove;
-        Fun.protect
+        Common.protect
```

Comment:
Do we need to change all of Fun.protect to Common.protect or the only one that really matters is this one 
because we're playing with Timeout here.

Reply:
The one I recorded happened while evaluating a `metavariable-pattern`. Any `protect` that can be reached from the body of `Time_limit.set_timeout` can potentially cause problems.